src/Disks/ObjectStorages/MetadataStorageFromPlainRewritableObjectStorage.cpp

```diff
@@ -178,7 +178,7 @@ void MetadataStorageFromPlainRewritableObjectStorage::load(bool is_initial_load)
                         }
 
                         /// Check that the file is a direct child.
-                        if (remote_file_path.substr(prefix_length) == filename)
+                        if (remote_file_path.substr(full_prefix_length) == filename)
```

Comment:
```suggestion
                        chassert(full_prefix_length < remote_file_path.size());
                        if (std::string_view(remote_file_path.data() + full_prefix_length) == filename)
```

A minor change, using string_view to avoid creating a substring.